### PR TITLE
PT-1204-rari-mine

### PR DESCRIPTION
**Pull request by SKonstantinS**
Ask for review contract RariMine

### CODE REVIEW

File: staking/contracts/RaRiMine.sol [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        require(stakeAmount > 0, "Amount for stake == 0");
+        balances[_msgSender()] = 0;
+        uint slope = stakeAmount;
+        if (slopePeriod == 0){
+            if (cliffPeriod == 0){
+                require(
+                token.transferFrom(tokenOwner, msg.sender, stakeAmount),
+                "Recipient transfer token error"
+                );
+                return;
```
Comment (disenotov) [inline staking/contracts/RaRiMine.sol:64]: no event

File: staking/contracts/RaRiMine.sol [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import "./LibStakingMath.sol";
+
+    /**
+     * @dev RariMine, a contract that allows you to increase (function plus())
+     *      or decrease (function minus()) the balance of ERC20 users tokens.
+     *      Also user can initiate stake or withdraw his own amount of ERC20 tokens.
+     *      In function claim() Withdraw case works only for slopePeriod == cliffPeriod == 0,
+     *      else stake`ll be initiate with amount equal ERC20 tokens user balance.
+     */
+contract RariMine is Ownable {
```
Comment (disenotov) [inline staking/contracts/RaRiMine.sol:18]: maybe make it upgradeable?

File: staking/test/RariMine.test.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		tokenOwner = accounts[2];
+		token = await ERC20.new();
+		staking = await Staking.new();
+		await staking.__Staking_init(token.address); //initialize staking, set token
+		rariMine = await RariMine.new(token.address, tokenOwner, staking.address); //initialize rariMine
+	})
+
+	describe("Check RariMine claim", () => {
+
+		it("Test 1. Claim() and stake tokens", async () => {
```
Comment (disenotov) [inline staking/test/RariMine.test.js:32]: formatting

File: staking/contracts/RaRiMine.sol (hunk 1 above)
Comment (SKonstantinS) [reply staking/contracts/RaRiMine.sol:65]: agree, done

File: staking/test/RariMine.test.js (hunk 3 above)
Comment (SKonstantinS) [reply staking/test/RariMine.test.js]: agree, done

File: staking/contracts/RaRiMine.sol (hunk 2 above)
Comment (SKonstantinS) [reply staking/contracts/RaRiMine.sol:18]: agree, done

**Pull request closed without merging**